### PR TITLE
Refactor waves

### PR DESCRIPTION
**Pull request by tanwayne890**
Code refactoring:
- Change `rounds` to `waveCounter`
- Rename GameManager.cs into GameStateManager.cs 
- Move `waveCounter` from GameStateManager.cs to WaveSpawner.cs
- change to egyptian brackets on some files

### CODE REVIEW

**Comment by tanwayne890** [comment]
forgot to make changes for another shader scene (shaderscene 16-10)

File: Assets/Scripts/GameManager/GameoverManager.cs [hunk 1]
```diff
@@ -3,23 +3,20 @@
 using UnityEngine.SceneManagement;
 using System.Collections;
 
-public class GameoverManager : MonoBehaviour
-{
+public class GameoverManager : MonoBehaviour {
```
Comment (trxe) [inline Assets/Scripts/GameManager/GameoverManager.cs]: nit: GameOver

File: Assets/Scripts/GameManager/GameoverManager.cs [hunk 2]
```diff
@@ -3,23 +3,20 @@
...
     public Text wavesText;
     public GameObject raycastOccluder;
 
     // everytime this is enabled
```
Comment (trxe) [inline Assets/Scripts/GameManager/GameoverManager.cs]: Replace with explanation of what is triggered OnEnable

File: Assets/Scripts/GameManager/GameoverManager.cs [hunk 3]
```diff
@@ -3,23 +3,20 @@
...
 
         raycastOccluder.SetActive(true);
         //Time.timeScale = 0f;
         StartCoroutine(PauseGame());
     }
 
-    IEnumerator PauseGame()
-    {
+    IEnumerator PauseGame() {
         //yield on a new YieldInstruction that waits for seconds.
```
Comment (trxe) [inline Assets/Scripts/GameManager/GameoverManager.cs]: actually why did you need the 1.5s wait?

File: Assets/Scripts/GameManager/GameoverManager.cs [hunk 4]
```diff
@@ -32,15 +29,13 @@ IEnumerator PauseGame()
 
     }
 
-    public void Retry() 
-    {
+    public void Retry() {
```
Comment (trxe) [inline Assets/Scripts/GameManager/GameoverManager.cs:31]: document what is re-enabled

File: Assets/Scripts/GameManager/WaveSpawner.cs [hunk 5]
```diff
@@ -3,92 +3,76 @@
...
+    void buttonOnClick() {
         isSkipWaveCountdownButtonVisible = false;
 
-        // reset timer 
-        countdownTimer = 0f;
+        countdownTimer = 0f; // reset timer 
         UpdateTimerIndicator(countdownTimer);
     }
 
     // Update is called once per frame
```
Comment (trxe) [inline Assets/Scripts/GameManager/WaveSpawner.cs]: remove

File: Assets/Scripts/GameManager/WaveSpawner.cs [hunk 6]
```diff
@@ -3,92 +3,76 @@
...
-    void Update()
-    {
+    void Update() {
         skipWaveCountdownButton.gameObject.SetActive(isSkipWaveCountdownButtonVisible);
         enemiesLeftText.text = string.Format("{0}", numEnemiesLeftInWave);
+        wavesCounterUI.text = "Wave " + string.Format("{0}", wavesCounter);
 
         // stops when first time seeing tutorial
```
Comment (trxe) [inline Assets/Scripts/GameManager/WaveSpawner.cs]: documentation: change to "Pause if player is viewing tutorial"

File: Assets/Scripts/GameManager/WaveSpawner.cs [hunk 7]
```diff
@@ -3,92 +3,76 @@
...
 
-        // win the game
-        if (waveIndex == waves.Length)
-        {
-            GetComponent<GameManager>().WinGame();
+
+        if (waveIndex == waves.Length) {
+            GetComponent<GameStateManager>().WinGame();
 
             // disable this script
```
Comment (trxe) [inline Assets/Scripts/GameManager/WaveSpawner.cs]: docs: change to "Disable WaveSpawner"

File: Assets/Scripts/GameManager/WaveSpawner.cs [hunk 8]
```diff
@@ -97,20 +81,17 @@ void Update()
...
         waveCountdownIndicator.maxValue = (int)(timeBetweenWaves * 100);
     }
 
-    // can pause the func execution
-    void SpawnWave()
-    {
-        //keep track of how many rounds survive
-        GameManager.rounds++;
+    void SpawnWave() {
+        //keep track of how many waves survive
```
Comment (trxe) [inline Assets/Scripts/GameManager/WaveSpawner.cs]: docs: remove

File: Assets/Scripts/Player/PlayerHealth.cs [hunk 9]
```diff
@@ -41,6 +41,6 @@ public class PlayerHealth : MonoBehaviour {
     void Die() {
         gameObject.SetActive(false);
         // TODO: add death animation?
```
Comment (trxe) [inline Assets/Scripts/Player/PlayerHealth.cs]: is this still needed?

File: Assets/Scripts/Player/PlayerHealth.cs [hunk 10]
```diff
@@ -11,13 +11,13 @@ public class PlayerHealth : MonoBehaviour {
     private float maxHealth = 100;
 
     private Image healthBar;
-    private GameManager gameManager;
+    private GameStateManager gameStateManager;
 
     // Start is called before the first frame update
```
Comment (trxe) [inline Assets/Scripts/Player/PlayerHealth.cs]: remove

File: Assets/Scripts/Player/PlayerHealth.cs [hunk 11]
```diff
@@ -11,13 +11,13 @@ public class PlayerHealth : MonoBehaviour {
...
     void Start() {
         healthBar = GameObject.Find("HealthCanvas/HealthBG/HealthBar").GetComponent<Image>();
         healthAmount = maxHealth;
-        gameManager = GameObject.Find("GameMaster").GetComponent<GameManager>();
+        gameStateManager = GameObject.Find("GameMaster").GetComponent<GameStateManager>();
     }
 
     // Update is called once per frame
```
Comment (trxe) [inline Assets/Scripts/Player/PlayerHealth.cs]: remove

File: Assets/Scripts/UI/PauseMenu.cs [hunk 12]
```diff
@@ -3,55 +3,46 @@
 using UnityEngine;
 using UnityEngine.SceneManagement;
 
-public class PauseMenu : MonoBehaviour
-{
+public class PauseMenu : MonoBehaviour {
     public static bool GameIsPaused;
```
Comment (trxe) [inline Assets/Scripts/UI/PauseMenu.cs:7]:
```suggestion
    public static bool gameIsPaused;
```

File: Assets/Scripts/UI/PauseMenu.cs [hunk 13]
```diff
@@ -3,55 +3,46 @@
...
 
     public GameObject pauseMenuUI;
     public GameObject gameplayCanvas;
     public GameObject raycastOccluder;
 
     // Update is called once per frame
```
Comment (trxe) [inline Assets/Scripts/UI/PauseMenu.cs]: remove

**Review by trxe — CHANGES_REQUESTED** [review]
notes on documentation and some unnecessary comments, and one variable to change name of

File: Assets/Scripts/GameManager/GameoverManager.cs (hunk 3 above)
Comment (tanwayne890) [reply Assets/Scripts/GameManager/GameoverManager.cs]: 1.5s wait is to show the full camera shake

Comment (tanwayne890) [reply Assets/Scripts/GameManager/GameoverManager.cs]: will refactor this PauseGame() again

File: Assets/Scripts/GameManager/WaveSpawner.cs (hunk 7 above)
Comment (tanwayne890) [reply Assets/Scripts/GameManager/WaveSpawner.cs]: i think can just remove docs, code is straight forward

File: Assets/Scripts/Player/PlayerHealth.cs (hunk 9 above)
Comment (tanwayne890) [reply Assets/Scripts/Player/PlayerHealth.cs]: TODO added by @Fenway17, i don't think is needed anymore, since we already end the game when player death

Comment (tanwayne890) [reply Assets/Scripts/Player/PlayerHealth.cs]: Will remove this in the same file

```diff
    public void TakeDamage(int amount) {
        healthAmount -= amount;

        // float number between 0 and 1
        healthBar.fillAmount = healthAmount / maxHealth;

-        // TODO: add damage animation?

        if (healthAmount <= 0) {
            Die();
        }
    }

```

File: Assets/Scripts/Player/PlayerHealth.cs (hunk 11 above)
Comment (tanwayne890) [reply Assets/Scripts/Player/PlayerHealth.cs]: i will just remove Update() since it is doing nothing

Comment (tanwayne890) [reply Assets/Scripts/Player/PlayerHealth.cs]: `Update is called once per frame` and `Start is called before the first frame update` seem to be auto-generated when we create new scripts. Most of the scripts will have these docs.

Is there any way to remove these docs when Start() and Update() are auto generated?

File: Assets/Scripts/UI/PauseMenu.cs (hunk 12 above)
Comment (tanwayne890) [reply Assets/Scripts/UI/PauseMenu.cs:7]: I could not find the naming convention for static variable from https://google.github.io/styleguide/csharp-style.html. But I can see an example from the style guide:

image

The first char in capital is better. WDYT?

File: Assets/Scripts/GameManager/GameoverManager.cs (hunk 2 above)
Comment (tanwayne890) [reply Assets/Scripts/GameManager/GameoverManager.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/ed2e671d7043215b75f3e44a9a90c87b33f58ca8

File: Assets/Scripts/GameManager/GameoverManager.cs (hunk 3 above)
Comment (tanwayne890) [reply Assets/Scripts/GameManager/GameoverManager.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/ef53215bb12b0f521d8b41d8ea826e51ae7d75df

File: Assets/Scripts/GameManager/GameoverManager.cs (hunk 1 above)
Comment (tanwayne890) [reply Assets/Scripts/GameManager/GameoverManager.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/89b4bab9dc49fe72b45664e99dd48e103565e89e

File: Assets/Scripts/GameManager/GameoverManager.cs (hunk 4 above)
Comment (tanwayne890) [reply Assets/Scripts/GameManager/GameoverManager.cs:31]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/44d9fa6016a87e1dbdf2b67f0e9d7a1fe4f029d4

File: Assets/Scripts/GameManager/WaveSpawner.cs (hunk 6 above)
Comment (tanwayne890) [reply Assets/Scripts/GameManager/WaveSpawner.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/322f86fb204bb5310456c7da0eabbae14c4136e4

File: Assets/Scripts/UI/PauseMenu.cs (hunk 13 above)
Comment (tanwayne890) [reply Assets/Scripts/UI/PauseMenu.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/0957f656dacd8bc2ddd92f330d94797101b5959d

File: Assets/Scripts/GameManager/WaveSpawner.cs (hunk 7 above)
Comment (tanwayne890) [reply Assets/Scripts/GameManager/WaveSpawner.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/c4f888f12c59275371101f75e4e06cfe2f3a2cc2

File: Assets/Scripts/GameManager/WaveSpawner.cs (hunk 8 above)
Comment (tanwayne890) [reply Assets/Scripts/GameManager/WaveSpawner.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/146b2e2fa8788a1d5c94402a581129e588ec68f8

File: Assets/Scripts/Player/PlayerHealth.cs (hunk 9 above)
Comment (tanwayne890) [reply Assets/Scripts/Player/PlayerHealth.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/18aaad4ce77ae7b0d5625d6af00ce7d1f21eb202

File: Assets/Scripts/Player/PlayerHealth.cs (hunk 11 above)
Comment (tanwayne890) [reply Assets/Scripts/Player/PlayerHealth.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/0ff714ecfe4ecc096035e98760d0bf47dc0a9dae

File: Assets/Scripts/Player/PlayerHealth.cs (hunk 10 above)
Comment (tanwayne890) [reply Assets/Scripts/Player/PlayerHealth.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/0ff714ecfe4ecc096035e98760d0bf47dc0a9dae

File: Assets/Scripts/GameManager/WaveSpawner.cs (hunk 5 above)
Comment (tanwayne890) [reply Assets/Scripts/GameManager/WaveSpawner.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/2d6d0d8f6feb24717b0a81a31e5d1ab608c57724

File: Assets/Scripts/UI/PauseMenu.cs (hunk 12 above)
Comment (trxe) [reply Assets/Scripts/UI/PauseMenu.cs:7]: ah ok then we can keep it that way

File: Assets/Scripts/GameManager/GameoverManager.cs (hunk 1 above)
Comment (trxe) [reply Assets/Scripts/GameManager/GameoverManager.cs]: Sorry for confusing, I meant GameOverManager

File: Assets/Scripts/Player/PlayerHealth.cs (hunk 11 above)
Comment (trxe) [reply Assets/Scripts/Player/PlayerHealth.cs]: I think must manually remove each time we create a new script. Don't have to deliberately go and remove from all rn, but if we update a file that has we should remove.

File: Assets/Scripts/GameManager/GameoverManager.cs (hunk 2 above)
Comment (trxe) [reply Assets/Scripts/GameManager/GameoverManager.cs]: I think it's better to write "Enabled when game is running"

**Review by trxe — APPROVED** [review]
Once the remaining 2 changes are made you can merge in

File: Assets/Scripts/GameManager/GameoverManager.cs (hunk 1 above)
Comment (tanwayne890) [reply Assets/Scripts/GameManager/GameoverManager.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/c87f8363539678c609970e6415c1565ce4437eb0

File: Assets/Scripts/GameManager/GameoverManager.cs (hunk 2 above)
Comment (tanwayne890) [reply Assets/Scripts/GameManager/GameoverManager.cs]: https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/200/commits/aee2e69515f7baa11755671b7f689d3ca11f7c68

**Pull request merged**